trace/trace.go

```diff
@@ -158,11 +158,12 @@ func startSpanInternal(name string, hasParent bool, parent SpanContext, remotePa
 	span := &Span{}
 	span.spanContext = parent
 	mu.Lock()
+	// TODO(jbd): Reduce the contention.
 	if !hasParent {
 		span.spanContext.TraceID = newTraceIDLocked()
 	}
 	span.spanContext.SpanID = newSpanIDLocked()
-	sampler := defaultSampler
+	sampler := config.DefaultSampler
```

Comment:
Is this thread safe? Shouldn't you use something like this: https://golang.org/pkg/sync/atomic/#Value?

Reply:
This whole block is guarded by a mutex right now.

If you are asking about the reference, in Go, things are pass by value. User cannot mutate the config.DefaultSampler.

Reply:
If you prefer atomic values for all the global state, I added a TODO for that.